projects/templates/src/lib/components/po-page-dynamic-table/interfaces/po-page-dynamic-table-actions.interface.ts

```diff
@@ -54,19 +54,65 @@ export interface PoPageDynamicTableActions {
   /**
    * @description
    *
-   * Rota criar um novo recurso, caso seja preenchida sera exibido uma ação no topo da página.
+   * Rota ou função para criar um novo recurso, caso seja preenchida sera exibido uma ação no topo da página.
    *
    * ```
    * actions = {
    *   new: 'new'
    * };
    * ```
    */
-  new?: string;
+  new?: string | Function;
 
   /** Habilita a ação de exclusão na tabela. */
   remove?: boolean;
 
   /** Habilita a ação de exclusão em lote na página. */
   removeAll?: boolean;
+
+  /**
+   * @description
+   *
+   *  Método/url que deve ser chamado antes da ação de inclusão
+   *  A url será chamada via POST.
+   *
+   * Tanto a função como a API devem retornar um objeto com a seguinte definição:
+   *
+   * ```
+   *  newUrl: string com nova rota para inserção, deve substituir a função ou rota definida anteriormente,
+   *  allowAction: boolean que define se deve ou não executar a ação de inserção (new)
+   * ```
+   *
+   * ```
+   * // exemplo de retorno
+   * {
+   *  newUrl: '/other-new-route',
+   *  allowAction: true
+   * }
+   * ```
+   *
+   * Caso o desenvolvedor queira que apareça alguma mensagem nessa ação ele pode criá-la na função chamada pela beforeNew
+   * ou definir a mensagem no atributo _message na resposta da api conforme definido em https://po-ui.io/guides/api#successMessages
```

Comment:
Sugestão de documentação
    - Equalizar os URLs e APIs. (L 76/77/95)
    - Ponto final, pois no portal o texto continuou do lado. (L 76/77)
    - Correção do _messages e realça-lo
    - Colocado o link da API em um texto para que fique melhor a visualização e acessivel.

```suggestion
   *  Método/URL que deve ser chamado antes da ação de inclusão.
   *  A URL será chamada via POST.
   *
   * Tanto a função como a API devem retornar um objeto com a seguinte definição:
   *
   * ```
   *  newUrl: string com nova rota para inserção, deve substituir a função ou rota definida anteriormente,
   *  allowAction: boolean que define se deve ou não executar a ação de inserção (new)
   * ```
   *
   * ```
   * // exemplo de retorno
   * {
   *  newUrl: '/other-new-route',
   *  allowAction: true
   * }
   * ```
   *
   * Caso o desenvolvedor queira que apareça alguma mensagem nessa ação ele pode criá-la na função chamada pela **beforeNew**
   * ou definir a mensagem no atributo `_messages` na resposta da API conforme definido em [Guia de implementação de APIs](https://po-ui.io/guides/api#successMessages)
```